### PR TITLE
ARROW-4297: [C++] Fix build error with MinGW-w64 32-bit

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

**Comment by kszucs** [comment]
How could We test this in an automatized way?

**Comment by kou** [comment]
I'm working: https://github.com/kou/arrow/tree/appveyor-msys :-)

File: cpp/src/arrow/util/sse-util.h [hunk 1]
```diff
@@ -110,7 +110,12 @@ static inline uint32_t SSE4_crc32_u32(uint32_t crc, uint32_t v) {
 }
 
 static inline uint32_t SSE4_crc32_u64(uint32_t crc, uint64_t v) {
+#if defined(__MINGW32__) && !defined(__MINGW64__)
+  DCHECK(false) << "MinGW-w64 32-bit doesn't support _mm_crc32_u64()";
```
Comment (pitrou) [inline cpp/src/arrow/util/sse-util.h]: Did you run the tests or are you only concerned about building? This is pretty much not the right solution.

**Review by pitrou — CHANGES_REQUESTED** [review]

**Comment by pitrou** [comment]
The appropriate solution is something like this, though according to @jeroen it is insufficient:
https://gist.github.com/pitrou/f4282f68a8dc9b11fe931bd0eeb1f8af

**Comment by jeroen** [comment]
@pitrou your [patch](https://gist.github.com/pitrou/f4282f68a8dc9b11fe931bd0eeb1f8af) will probably work icw/ this PR. But you really need to put the call to `_mm_crc32_u64` behind a macro otherwise it won't compile (even though you're not actually using that function).

**Comment by pitrou** [comment]
@jeroen Right. Can you open a PR for that?

File: cpp/src/arrow/util/sse-util.h (hunk 1 above)
Comment (kou) [reply cpp/src/arrow/util/sse-util.h]: This is just fixing build error.
I'm working on test support on MSYS2 in separated issues.
I'll be back again after we can run tests on MSYS2.

**Comment by kou** [comment]
Done by #3693.

**Pull request closed without merging**